dlk/python/dlk/templates/src/func/impl/pack_16bit.cpp

```diff
@@ -0,0 +1,47 @@
+/* Copyright 2019 The Blueoil Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "func/impl/pack_16bit.h"
+#include <cassert>
+
+namespace dlk {
+
+namespace impl {
+
+void pack_16bit(const BIN_CONV_OUTPUT input[], QUANTIZED_PACKED output[], const std::size_t length) {
```

Comment:
Other thing is that output could contain number of channels which is not multiple of 32, while your code seems to assuming that everything is multiple of 32, as otherwise you need to know not only lenght, but also how many channels you have.

For example:
If number of channel is 4, and first 8 values are 3,3,3,3,1,1,1,1, and all other values are 0 output proper output should be
```
LSB[0] = 0x0000000F // 4 channels packed
MSB[0] = 0x0000000F
LSB[1] = 0x0000000F // 4 channels packed
MSB[1] = 0x00000000
```
While your code seems to be producing something like
```
LSB[0] = 0x000000FF // 32 channels packed
MSB[0] = 0x0000000F
```
for this case... :thinking:

Reply:
In my code, number of channels is ceiling up to multiple of 32.